ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,387 @@
+    # Manifold in which all the decision variables leave
+    manifold::Union{Nothing,ManifoldsBase.AbstractManifold}
+    # Description of the problem in Manopt
+    problem::Union{Nothing,Manopt.AbstractManoptProblem}
+    # State of the optimizer
+    state::Union{Nothing,Manopt.AbstractManoptSolverState}
+    # Starting value for each variable
+    variable_primal_start::Vector{Union{Nothing,Float64}}
+    # Sense of the optimization, e.g., min, max or no objective
+    sense::MOI.OptimizationSense
+    # Model used to compute gradient of the objective function with AD
+    nlp_model::MOI.Nonlinear.Model
+    # Solver parameters set with `MOI.RawOptimizerAttribute`
+    options::Dict{String,Any}
+    function Optimizer()
+        return new(
+            nothing,
+            nothing,
+            nothing,
+            Union{Nothing,Float64}[],
+            MOI.FEASIBILITY_SENSE,
+            MOI.Nonlinear.Model(),
+            Dict{String,Any}(),
+        )
+    end
+end
+
+MOI.get(::Optimizer, ::MOI.SolverVersion) = "0.4.37"
+
+function MOI.is_empty(model::Optimizer)
+    # TODO replace `isnothing(model.nlp_model.objective)`
+    #      by `MOI.is_empty` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    return isnothing(model.manifold) &&
+           isempty(model.variable_primal_start) &&
+           isnothing(model.nlp_model.objective) &&
+           model.sense == MOI.FEASIBILITY_SENSE
+end
+
+function MOI.empty!(model::Optimizer)
+    model.manifold = nothing
+    model.problem = nothing
+    model.state = nothing
+    empty!(model.variable_primal_start)
+    model.sense = MOI.FEASIBILITY_SENSE
+    # TODO replace by `MOI.empty!` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    model.nlp_model.objective = nothing
+    return nothing
+end
+
+"""
+    MOI.supports(::Optimizer, attr::MOI.RawOptimizerAttribute)
+
+Return a `Bool` indicating whether `attr.name` is a valid option name
+for `Manopt`.
+"""
+function MOI.supports(::Optimizer, ::MOI.RawOptimizerAttribute)
+    # FIXME It should depend on `attr.name`
+    return true
+end
+
+function MOI.get(model::Optimizer, attr::MOI.RawOptimizerAttribute)
+    if !MOI.supports(model, attr)
+        throw(MOI.UnsupportedAttribute(attr))
+    end
+    return model.options[attr.name]
+end
+
+function MOI.set(model::Optimizer, attr::MOI.RawOptimizerAttribute, value)
+    if !MOI.supports(model, attr)
+        throw(MOI.UnsupportedAttribute(attr))
+    end
+    model.options[attr.name] = value
+    return nothing
+end
+
+MOI.get(::Optimizer, ::MOI.SolverName) = "Manopt"
+
+"""
+    MOI.supports_incremental_interface(::Optimizer)
+
+Return `true` indicating that `Manopt.Optimizer` implements
+`MOI.add_constrained_variables` and `MOI.set` for
+`MOI.ObjectiveFunction` so it can be used with `JuMP.direct_model`
+and does not require a `MOI.Utilities.CachingOptimizer`.
```

Comment:
Could we maybe link variables to your docs? Or are the links too fragile?

Because just from this text I do not understand much what this indicates.